src/main/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAO.java

```diff
@@ -398,11 +398,18 @@ public void executePatchDataset(Handle handle,
       Integer userId,
       List<DatasetProperty> properties) {
     // update dataset
-    datasetDAO.updateDatasetNameWithUpdateUser(
-        datasetId,
-        datasetName,
-        new Timestamp(new Date().getTime()),
-        userId);
+    if (datasetName == null || datasetName.isBlank()) {
```

Comment:
When I see validation like `isBlank()` in a DAO, I wonder if it's actually possible for a blank string to get here, and if so, if it could be validated further up the call chain. My general approach is to validate data at the outermost layer (here it would be in `DatasetResource`). This can help avoid duplication of validation at different downstream locations. The validation could replace blank with `null`, or it could overwrite blank and null with the current dataset name.

Reply:
In this case, we have validation on the patch object itself. I'll add some more there for additional coverage.

Reply:
And to be clear, the patch validation methods are called from `DatasetResource`

Reply:
The problem I see here is that the validation isn't normalizing the data, so this code needs to check for two equivalent cases instead of one case. If the validation folded equivalent cases into a single canonical case then you'd only need to check one case here.